svea_vision/README.md

```diff
@@ -1,11 +1,91 @@
 # svea_vision
+This package contains the vision system for SVEA. It contains nodes for object detection, pose estimation, aruco markers and sidewalk segmentation. 
 
 ## Installation
 
-Please refer to [svea](https://github.com/KTH-SML/svea) on how to install and
-run SVEA software.
+Please refer to [svea](https://github.com/KTH-SML/svea) on how to install and run SVEA software.
 
-## Usage
+## ROS Nodes
 
-This README file is meant to be replaced by something relevant to the package.
+### object_detect.py
+TODO
 
+### object_pose.py
+TODO
+
+### aruco_detect.py
+TODO
+
+### pedestrian_flow_estimate.py
+
+This ROS node, `pedestrian_flow_estimate.py`, subscribes to topics that provide detected pedestrian data, and publishes estimated speed and acceleration of these pedestrians.
+The main purpose of this node is to track and predict the state of each object detected by a camera system using moving average filters. The node processes pedestrian positions, calculates velocities through numerical differentiation, and then smoothes these velocities using a moving average filter. The filtered velocity is used to compute acceleration, which is also filtered to produce more accurate estimates.
+
+#### Subscribed Topics
+
+- `/detection_splitter/persons` (`svea_vision_msgs/StampedObjectPoseArray`): This topic contains data about detected persons, including their positions and IDs.
+
+#### Published Topics
+
+- `~float_1` (`std_msgs/Float64`): Publishes raw acceleration values (ay) for debugging purposes.
+- `~float_2` (`std_msgs/Float64`): Publishes smoothed velocity values (vy) for debugging purposes.
+- `~pedestrian_flow_estimate` (`svea_vision_msgs/PersonStateArray`): Publishes the estimated state (position, velocity, acceleration) of each detected pedestrian.
+
+#### Parameters
+
+- `~discard_id_threshold` (float): Threshold to detect wrong pose estimates due to pedestrian boxes distortion/ bouncing detected position.
+- `~max_time_missing` (float): Time in seconds to drop an ID if no data is received.
+- `~vel_filter_window` (int): Window size for the velocity filter.
+- `~acc_filter_window` (int): Window size for the acceleration filter.
+
+#### Relevant Methods
```

Comment:
These comments should not be in the README, but rather in docstrings for each method/function